src/test/java/com/exasol/adapter/dialects/rls/QueryRuntimeIT.java

```diff
@@ -0,0 +1,113 @@
+                "INSERT INTO SIMPLE_SALES.ORDER_ITEM VALUES\n" //
+                        + "(1, 'John Smith', 'Pen', 3, 1),\n" //
+                        + "(1, 'John Smith', 'Paper', 100, 3),\n" //
+                        + "(1, 'John Smith', 'Eraser', 1, 7),\n" //
+                        + "(2, 'Jane Doe', 'Pen', 2, 2),\n" //
+                        + "(2, 'Jane Doe', 'Paper', 200, 1)", //
+                "CREATE TABLE SIMPLE_SALES.EXA_RLS_USERS(EXA_USER_NAME VARCHAR(128), EXA_ROLE_MASK DECIMAL(20,0))",
+                "INSERT INTO SIMPLE_SALES.EXA_RLS_USERS VALUES\n" //
+                        + "('SALES', 1),\n" //
+                        + "('DEVELOPMENT', 2),\n" //
+                        + "('FINANCE', 4)");
+    }
+
+    private static void uploadRlsAdapter() throws InterruptedException, BucketAccessException, TimeoutException {
+        final Path localAdapterPath = Path.of("target", ADAPTER_FILENAME).toAbsolutePath();
+        container.getDefaultBucket().uploadFile(localAdapterPath, ADAPTER_FILENAME);
+    }
+
+    private static void createRowLevelSecuritySchema() throws SQLException {
+        final Connection connection = container.createConnection("");
+        final Bucket bucket = container.getDefaultBucket();
+        executeQueries(connection, //
+                "CREATE SCHEMA RLS_SCHEMA", //
+                "CREATE OR REPLACE JAVA ADAPTER SCRIPT RLS_SCHEMA.RLS_VS_ADAPTER AS\n" //
+                        + "    %scriptclass com.exasol.adapter.RequestDispatcher;\n" //
+                        + "    %jar /buckets/" + bucket.getBucketFsName() + "/" + bucket.getBucketName() //
+                        + "/" + ADAPTER_FILENAME + ";\n" //
+                        + "/", //
+                "CREATE CONNECTION EXASOL_JDBC_CONNECTION TO 'jdbc:exa:localhost:" //
+                        + container.getExposedPorts().get(0) + "' USER '" + container.getUsername()
+                        + "' IDENTIFIED BY '" //
+                        + container.getPassword() + "'", //
+                "CREATE VIRTUAL SCHEMA RLS_VS \n" //
+                        + "    USING RLS_SCHEMA.RLS_VS_ADAPTER\n" //
+                        + "    WITH\n" //
+                        + "    SQL_DIALECT     = 'EXASOL_RLS'\n" //
+                        + "    CONNECTION_NAME = 'EXASOL_JDBC_CONNECTION'\n" //
+                        + "    SCHEMA_NAME     = 'SIMPLE_SALES'\n" //
+                        + "    IS_LOCAL        = 'true'");
+    }
+
+    private static void executeQueries(final Connection connection, final String... sqls) throws SQLException {
+        final Statement statement = connection.createStatement();
+        for (final String sql : sqls) {
+            LOGGER.info(sql);
+            statement.execute(sql);
+        }
+    }
+
+    // [itest->qs~total-runtime-of-secured-simple-query~1]
+    @Test
+    void testSimpleQueryRuntime() throws NoDriverFoundException, SQLException {
```

Comment:
This test fails both on Travis and on my machine. Does it work for you?
Error from my machine:  (a value less than or equal to <53L> or a value less than or equal to <1048L>)
     but: was <1714L>

Reply:
That is very sad. Please check on your machine how much of that time was spent inside the VS adapter.

Reply:
Raised to 2 seconds.